engine/src/main/java/org/hibernate/validator/metadata/descriptor/HibernateConstraintDescriptor.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.metadata.descriptor;
+
+import java.lang.annotation.Annotation;
+
+import javax.validation.metadata.ConstraintDescriptor;
+
+/**
+ * Extension to {@code ConstraintDescriptor} which provides functionality
+ * specific to Hibernate Validator.
+ *
+ * @author Guillaume Smet
+ * @since 6.0
+ */
+public interface HibernateConstraintDescriptor<T extends Annotation> extends ConstraintDescriptor<T> {
```

Comment:
No, it's not only for testing. We also need it for the implementation as we need a reference type and I don't think using the `javax.validation` one makes sense.

I don't think it costs us much to have it prepared.